portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -48,31 +54,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     int numComments = Integer.parseInt(getParameter(request, "comments", "5"));
     String language = getParameter(request, "language", "EN");
 
-    //Queries datastore for comments
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Comment").addSort("posted", SortDirection.DESCENDING);
-    PreparedQuery results = datastore.prepare(query);
-
-    //Converts result from query into array of comment objects
-    ArrayList<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(numComments))) {
-      String name = (String) entity.getProperty("name");
-      String body = (String) entity.getProperty("body");
-      Date posted = (Date) entity.getProperty("posted");
-      long votes = (long) entity.getProperty("votes");
-      double score = (double) entity.getProperty("score");
-      long id = entity.getKey().getId();
+    //Connect to cache
+    Cache cache;
+    try {
+      CacheFactory cacheFactory = CacheManager.getInstance().getCacheFactory();
+      cache = cacheFactory.createCache(Collections.emptyMap());
+    } catch (CacheException e) {
+      System.out.println("Cache error");
+      return;
+    }
 
-      //Translates comment into selected language
-      String translatedBody = translateComment(body, language);
+    //Retrieve array of comments
+    ArrayList<Comment> comments;
+    if (cache.containsKey("comments")){
+      comments = (ArrayList<Comment>) cache.get("comments");
+      System.out.println("Retrieved from cache");
```

Comment:
As a general rule, I think these will be better as log messages (check the example I sent for objectify). This way you have them at DEBUG level and so optionally turn them off.